### PR TITLE
Fixes #6404 jrubyc: mangled ("--target") path on Windows

### LINKED ISSUE #6404: jrubyc has issues with mangled (target) path on Windows
not 100% sure but seems like `jrubyc` is broken with `--target` path specified,
the path seems to end up being invalid on Windows, here's 2 (pended) reproducers: https://github.com/jruby/jruby/commit/851629a431b4c5b9fe63d78723cd32980f030c85

### PR DESCRIPTION
**Pull request by MariuszCwikla**
Fixes #6404

### CODE REVIEW

**Comment by kares** [comment]
Thanks Mariusz, :trophy: find!

**Review by kares — APPROVED** [review]

File: test/jruby/compiler/test_jrubyc.rb [hunk 1]
```diff
@@ -15,8 +16,8 @@ def setup; require 'jruby/jrubyc'
 
     @tempfile_stderr = Tempfile.open("test_jrubyc_stderr")
     @old_stderr = $stderr.dup
-    $stderr.reopen @tempfile_stderr
-    $stderr.sync = true
+#    $stderr.reopen @tempfile_stderr
+#    $stderr.sync = true
```
Comment (headius) [inline test/jruby/compiler/test_jrubyc.rb:20]: Perhaps we can just remove these lines?

**Review by headius — APPROVED** [review]
Looks good to me.

File: test/jruby/compiler/test_jrubyc.rb (hunk 1 above)
Comment (headius) [reply test/jruby/compiler/test_jrubyc.rb:20]: Actually I suspect these lines were commented out solely for debugging and should be restored. I'll take care of it.

**Pull request merged**